Framework/PythonInterface/plugins/algorithms/LRSubtractAverageBackground.py

```diff
@@ -59,10 +61,14 @@ def PyExec(self):
 
         sum_peak = self.getProperty("SumPeak").value
 
-        # Number of pixels in each direction
-        #TODO: revisit this when we update the IDF
-        number_of_pixels_x = int(workspace.getInstrument().getNumberParameter("number-of-x-pixels")[0])
-        number_of_pixels_y = int(workspace.getInstrument().getNumberParameter("number-of-y-pixels")[0])
+        detector = self.getProperty("TypeOfDetector").value
+        if detector == "LinearDetector":
+            number_of_pixels_x = 1
+            number_of_pixels_y = int(workspace.getNumberHistograms())
+        else:
+            # TODO: revisit this when we update the IDF
+            number_of_pixels_x = int(workspace.getInstrument().getNumberParameter("number-of-x-pixels")[0])
```

Comment:
I think here one should check if the workspace has instrument attached, and if it has the named parameter with `hasParameter` method.

Reply:
Good spot, I'll add that in